### ISSUE TITLE
Issue: Multiline string field in data view

### ISSUE THREAD

**Issue by davidsneighbour** [estruyf/vscode-front-matter#363]
I would expect the Dashboard > Data view to work according to [the docs for field configurations for strings](https://frontmatter.codes/docs/content-creation/fields#string), as in `single: false` and `wysiwyg: true` resulting in extended text fields. It doesn't though. 

Config:

```
  "frontMatter.data.files": [
    {
      "id": "music2program2",
      "title": "Music to program to",
      "file": "[[workspace]]/data/dnb/kollitsch/music.json",
      "fileType": "json",
      "labelField": "name",
      "schema": {
        "title": "music2program2",
        "type": "object",
        "required": [
          "name",
          "link"
        ],
        "properties": {
          "name": {
            "type": "string",
            "title": "Name"
          },
          "link": {
            "type": "string",
            "title": "URL"
          },
          "description": {
            "type": "string",
            "title": "Description",
            "single": false,
            "wysiwyg": true
          }
        }
      }
    }
  ],
```

results in:

![20220627-22:20:15-Selection-001](https://user-images.githubusercontent.com/83281/175975613-ffbe1dcb-0040-4af2-b191-d3581ae66e52.png)

It would be nice to have at least a multiline field here. Maybe even globally some form of distinction between string and "multiline string" as field type in general?

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04

**Comment by estruyf** [comment]
@davidsneighbour `single` and `wysiwyg` are not supported. The files are automatically rendered by the JSON schema and do not follow the same logic as the other content fields (yet). I'm going to tag this as an enhancement.

**Comment by davidsneighbour** [comment]
Ping :)

I just worked on my data file configuration and was about to open a new issue when I realized that I had already done so a while back.

The issues remain:

- from the view, there is no need to have "your $title data items" (it takes too much space and is understood from the left column). "Select your data type", "Items", and "Create or Modify data" would result in a single header line instead of 1 to 3 in the screenshot below.
- a multiline field would be really nice ;) my description in this specific case is a markdown text that might be long. 

new idea ;):
- validation. the slug field here should not have spaces or special characters, so it would be nice to maybe have some form of a regular expression to match the content of each field against. 

![20230313-21:50:06-Selection-001](https://user-images.githubusercontent.com/83281/224738425-685c1d26-f97f-490f-ab59-bb631911f2ea.png)

This still stays enhancement, so there is no pressure :) but I got ideas for the data functionality. Think recipes or "trading card games". I think it might be a powerful feature for data file enabled SSGs.

**Comment by michaeltlombardi** [comment]
I have a strong need for multiline fields as well (also using markdown in those fields).

**Comment by estruyf** [comment]
I heard both of you 😀

![image](https://user-images.githubusercontent.com/2900833/224819676-b428a355-1bd1-4dad-a3bd-e295238d527d.png)

To support this, I must add a new property for the long text (textarea) field. 

```json
{
  "name": {
    "type": "string",
    "title": "Name"
  },
  "url": {
    "type": "string",
    "title": "URL"
  },
  "description": {
    "type": "string",
    "longtext": true,
    "title": "Description"
  }
}
```

Above I used `longtext` (as that is the field to load in uniforms, but if you have a better suggestion, let me know.

**Comment by michaeltlombardi** [comment]
That's fine! As long as it's documented, I can work with it (`multiline` might be a more intuitive key name but I'm not too fussed about it).

**Comment by estruyf** [comment]
and so it became `multiline`.

@davidsneighbour @michaeltlombardi you can configure it as follows:

```json
{
  "name": {
    "type": "string",
    "title": "Name"
  },
  "url": {
    "type": "string",
    "title": "URL"
  },
  "description": {
    "type": "string",
    "multiline": true, // Set the multiline support
    "title": "Description"
  }
}
```

**Comment by davidsneighbour** [comment]
The label of the field is not fat like the other fields. All else is as I dreamed of :)

![20230314-23:01:40-Selection-001](https://user-images.githubusercontent.com/83281/225064544-85acf53e-acbd-43cb-b40d-ccb8dfce3942.png)

**Comment by estruyf** [comment]
Good catch, just rendering the OOTB experience, didn't notice it, will fix it.

**Comment by estruyf** [comment]
Just released a fix for this @davidsneighbour
